src/compiler/xslt/compile/compile-scenario.xsl

```diff
@@ -329,4 +331,16 @@
       <xsl:call-template name="x:compile-child-scenarios-or-expects" />
    </xsl:template>
 
+   <xsl:template name="x:set-up-context" as="element()+">
+      <xsl:param name="context" as="element(x:context)?" required="yes"/>
```

Comment:
It seems this can be a tunnel parameter:

```suggestion
      <xsl:param name="context" as="element(x:context)" tunnel="yes"/>

```

and then `<xsl:with-param name="context" select="$context"/>` can be removed from `<xsl:call-template name="x:set-up-context">`.
What do you think?

Reply:
@AirQuick , all your review suggestions are good ideas, and I've addressed them in 7af7ae68ddcbdfe64043c56cdd4c6fbb87a91918 . Thanks!